### DISCUSSION TITLE
react-spring doesn't list peerDependencies

### DISCUSSION THREAD

**Discussion by MarcCelani-at** [pmndrs/react-spring#1793] [Support]
## 🐛 Bug Report

react-spring doesn't list peerDependencies, which causes these error in Yarn3:

```
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide konva (p18128), requested by @react-spring/konva
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react (p95902), requested by @react-spring/konva
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react (p5bc6b), requested by @react-spring/native
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react (p8b4fa), requested by @react-spring/three
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react (pb7c2a), requested by @react-spring/web
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react (p5e33c), requested by @react-spring/zdog
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react-dom (p42840), requested by @react-spring/web
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react-dom (p10810), requested by @react-spring/zdog
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react-konva (pf9dae), requested by @react-spring/konva
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react-native (pd1e8c), requested by @react-spring/native
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react-three-fiber (p12675), requested by @react-spring/three
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide react-zdog (pc20b7), requested by @react-spring/zdog
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide three (p19cd4), requested by @react-spring/three
➤ YN0002: │ react-spring@npm:9.0.0-rc.3 doesn't provide zdog (p98401), requested by @react-spring/zdog
```

## To Reproduce

1. Create a new package using yarn3.
2. Run yarn install

## Expected behavior

No missing dependency warnings

## Environment

- `react-spring` v9.0.0-rc.3

**Comment by joshuaellis** [comment]
Personally I would not install `react-spring`, I would instead, install a specific target e.g `@react-spring/web` which therefore you need to install `react-dom`.

**Reply by MarcCelani-at** [reply]
That sounds like a good idea, I'll take a look at that. But this still feels like a bug to me, since the react-spring package does exist.

**Reply by joshuaellis** [reply]
it's not a bug, you have to install the peer-deps, they're optional so _if_ you do want install `react-spring` you just need to pick which target you want to use. It's a catch-all package and because of that, it has to have `optional peer-deps`

**Reply by MarcCelani-at** [reply]
I'm not sure, but I think what yarn is trying to do here is encourage all the intermediate projects to list peer dependencies. That is, when a project depends on dependency D1, which in turn depends on dependency D2, and D2 lists a peer dependency P, then D1 should also list P as a peer dependency so as to signal to the projects that depend on D1 that you should install that peer dep.

**Reply by MarcCelani-at** [reply]
Unless of course D1 directly depends on P in which case the peer dependency is met.
